### ISSUE TITLE
Evaluation FEFunctions on embedded meshes

### ISSUE THREAD

**Issue by MiroK** [gridap/Gridap.jl#876]
Hi, below I am trying to evaluate a finite element function at a point where the function is in a FE space over on embedded mesh. The function can be created (and eg. plotted) but there is an error when I want to do interpolation at a point (which is 
in the mesh). 

```julia
using Gridap

domain = (0, 1, 0, 1)
partition = (5, 5)
mesh = CartesianDiscreteModel(domain, partition)
dmesh = BoundaryTriangulation(mesh)

elm = ReferenceFE(lagrangian, Float64, 1)
W = FESpace(dmesh, elm)
fh = interpolate_everywhere(x -> x[1]+x[2], W)

fh(Point(0., 0.))   # Errors
```

From the error I suspect that the issue is due to the fact that the the point is in $R^2$ while the mesh of `fh` is topologically 1d.  
Can one use pseudo inverse in this case?
```
ERROR: LoadError: DimensionMismatch: matrix is not square: dimensions are (1, 2)
Stacktrace:
  [1] checksquare(A::Type)
    @ LinearAlgebra ~/Documents/Software/julia/usr/share/julia/stdlib/v1.8/LinearAlgebra/src/LinearAlgebra.jl:232
  [2] #s25#506
    @ ~/.julia/packages/StaticArrays/B0HhH/src/inv.jl:75 [inlined]
  [3] var"#s25#506"(S::Any, ::Any, #unused#::Any, A::Any)
    @ StaticArrays ./none:0
  [4] (::Core.GeneratedFunctionStub)(::Any, ::Vararg{Any})
    @ Core ./boot.jl:582
  [5] inv
    @ ~/.julia/packages/StaticArrays/B0HhH/src/inv.jl:5 [inlined]
  [6] inv(a::TensorValue{1, 2, Float64, 2})
    @ Gridap.TensorValues ~/.julia/packages/Gridap/OTENN/src/TensorValues/Operations.jl:483
```

**Comment by amartinhuertas** [comment]
> From the error I suspect that the issue is due to the fact that the the point is in while the mesh of fh is topologically 1d.

Yes, I agree. 

I think that the evaluation of FE Functions defined on a manifold on physical space points is just not possible in Gridap. While it might be implemented, I think we decided not to implement it as it may open the door for undetermined cases in which one may want to evaluate the funcion on points which are not spanned by the manifold. 

But I am not 100% sure, perhaps @oriolcg or @ericneiva or @fverdugo have a clearer picture on this than me.

**Comment by oriolcg** [comment]
Yes, that's the problem. We also faced this issue and, indeed, the solution is to use the pseudo inverse of the map. This was already implemented, but there is still a function that is not using it. @shagun751 was planing to do a PR with this fix soon.

As @amartinhuertas said, we had some discussion related to interpolation into manifolds. There are two issues here. First, if you can identify the point in the manifold, you need the pseudo inverse of a rectangular matrix (this should be fixed in the comin PR). The discussion we had, and didn't find a proper way of addressing the issue yet, is related to how to identify the point in the manifold. Now you can do that in Gridap (as you already do) by interpolating for instance to a point in the boundary, assuming that the boundary is on a plane aligned with the coordinate system. However, the interpolation algorithm doesn't know what's the coordinate normal to the plane of the boundary. For instance, you'll get the same solution if you do `fh(Point(0.0,0.0))` than `fh(Point (0.0, 10.0))` or `fh(Point(0.0))`, assuming your boundary is aligned with `x` coordinate.

One way of solving this is to be able to find points on parametrized manifolds, but this is not implemented.

I hope this explanation makes sense...

**Comment by MiroK** [comment]
Thank you very much for the responses. In our application we can use Nitsche method as a workaround but once the pseudo-inverse PR lands I'll be happy to give it a try.

**Comment by shagun751** [comment]
@oriolcg 
I have created a pull request #879 to resolve the error encountered while using `evaluate()` for a FieldFEFunction defined on a boundary.
